packages/typespec-client-generator-core/src/interfaces.ts

```diff
@@ -97,6 +97,7 @@ export type SdkType =
   | SdkArrayType
   | SdkTupleType
   | SdkDictionaryType
+  | SdkNullableType
```

Comment:
`SdkNullableType`: I'm a bit confused on the nullable type, I thought it is type whose kind is `null` to represent `null` in tsp. As the example in chat shows.

![image](https://github.com/Azure/typespec-azure/assets/87355844/9d00d1d5-ae93-4d4e-a79a-d0e175fd2e50)

Reply:
The final implementation changed to introduce a new `SdkNullableType` in TCGC instead of reusing the `SdkUnionType` to reduce complexity for emitters.

Reply:
Yeah, I realized the design changed from we have talked from chat. Thanks for the update.

I kind of feel that the current design looks similar as previous ‘sdktype.nullable’, as they both try to represent a nullable type, but current one introduce ‘SdkNullableType’ and brings another level of hierarchy/complexity. And the current design also breaks downstream.

Would like to know the reason why the design changed and the benefit of the current design (comparing to sdktype.nullable).